processor/src/decoder/mod.rs

```diff
@@ -268,6 +269,21 @@ impl Decoder {
         self.trace.program_hash()
     }
 
+    /// Returns an operation to be executed at the specified clock cycle. Only applicable in debug mode.
+    pub fn get_operation_at(&self, clk: usize) -> Operation {
+        self.debug_info.get_operation_at(clk)
+    }
+
+    /// Returns list of assembly operations in debug mode.
+    pub fn assembly_ops(&self) -> &[(usize, AssemblyOp)] {
+        self.debug_info.assembly_ops()
+    }
```

Comment:
Maybe instead of these two, just have `pub fn debug_info(&self) -> &DebugInfo`? If we do this, we could add `debug_assert` check if we are in debug mode or not. Basically, if we are trying to get debug info but we are not in debug mode, this would panic.

I'd still keep `in_debug_mode()` as a separate function though.

Reply:
Makes sense. Done.